MonoGame.Framework/Vector3.cs

```diff
@@ -522,7 +530,7 @@ public bool Equals(Vector3 other)
         /// <returns>Hash code of this <see cref="Vector3"/>.</returns>
         public override int GetHashCode()
         {
-            return (int)(this.X + this.Y + this.Z);
+            return X.GetHashCode() + Y.GetHashCode() + Z.GetHashCode();
```

Comment:
> I looked into XNA dll using dotPeek.

Then we have to close this PR and unfortunately can't accept any more submissions from you.

You cannot steal code or ideas from peeking at the XNA source code.